composer.lock

```diff
@@ -0,0 +1,254 @@
+{
```

Comment:
this file looks like the json file above but still but why .lock?

Reply:
This file is for tracking the exact version of the third-party dependency installed when I first added twig into composer.json. 
When you add any dependency composer will look for the best version of its dependency to satisfy your requirements. In my case, it "twig/twig": "^3.0". 
When you try to run `composer instal,` it will install the exact version of twig that pointed in the 'composer.lock', so our code will always be the same; even if twig releases a new version, you will get the same version as mine.